### ISSUE TITLE
Compliance/Compliance Obligation - External User can view operations compliance obligations

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-compliance#17]
#### Description:

As an external user, I can view a grid of compliance obligations for each of my operations

- [Design in Figma](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=20107-1332&t=f6iGz2pZS8IzlScu-1)

#### Acceptance Criteria:

- [ ] Given I'm on the Compliance Summaries tab, I can see a grid of each of my operations and their compliance obligation amounts.
    - The excess emissions is the emissions total pulled from reporting, displayed in tCO2e. If over the emissions limit, the excess emissions will be a positive integer. If under the emissions limit, this will show as a negative integer and the outstanding balance of 0tCO2e.The table will have the following 
    
- [ ] The grid contains the following columns:
Reporting year
Operation name
Excess emissions (shown in tCO2e)
Outstanding balance (shown in tCO2e)
Compliance status
Penalty status
Obligation ID
Action (has a clickable link)

- [ ]  Given I am on the grid, I can see a list of all reporting years (including past years), and the grid will show the compliance obligations for each operation for all years 
- [ ] Given I am on the Compliance Obligation Grid, I can see my COAM balance being automatically adjusted as my payments are being made.
    - If I edit and re-submit a report, it will show the new amount of excess emissions over or under the limit and the new outstanding balance and compliance status

- [ ] Given that I'm on the grid, each column is sortable (A-Z, Z-A)
- [ ] Given that I'm on the grid, each column has a free-text search box

[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=17825-2534&t=hoovKA1meA6RBOCl-4)

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies

- Blocked by
- Blocking

**Comment by patrickisaac** [comment]
Refinement Notes:
- @nanyangpro to update appropriate wireframes

**Comment by nanyangpro** [comment]
Link to design is updated in the ticket description

**Comment by vivanovbc** [comment]
I come up with following tasks for this ticket so we can create them separately:
- Compliance. Add base compliance summaries page [2]
- Compliance. Summaries. List compliance obligations with Reporting year, Operation name, Excess emissions [5]
- Compliance. Summaries. Add Outstanding balance column [TBD]
- Compliance. Summaries. Add Compliance status column [3]
- Compliance. Summaries. Add Penalty status column [TBD]
- Compliance. Summaries. Add Obligation ID column [1]
- Compliance. Summaries. Add base Action column [1]

**Comment by patrickisaac** [comment]
Newly created tickets can be found here:

Compliance. Add base compliance summaries page #109 
Compliance. Summaries. List compliance obligations with Reporting year, Operation name, Excess emissions #110 
Compliance. Summaries. Add Outstanding balance column #111 
Compliance. Summaries. Add Compliance status column #112 
Compliance. Summaries. Add Penalty status column #113 
Compliance. Summaries. Add Obligation ID column #114 
Compliance. Summaries. Add base Action column #115

**Comment by patrickisaac** [comment]
Refinement Notes:
- Closing this out as this information has been transferred into a milestone
- Sub-tickets created
